projects/ng-components/src/lib/controls/advanced-search/advanced-search.component.ts

```diff
@@ -88,16 +88,16 @@ export class AdvancedSearchComponent implements OnInit {
     if (this.selectedField.dataType === 'daterange') {
       if (!this.selectedField.multiTerm) {
         this.filters = this.filters.filter((f) => {
-          return f.member !== QueryParameters.FILTER_FROM && f.member !== QueryParameters.FILTER_TO;
+          return f.dataType !== 'datetime';
         });
       }
       if (this.fieldValueDateFrom) {
-        const filterClauseFrom = new FilterClause(QueryParameters.FILTER_FROM, this.fieldValueDateFrom, Operators.GREATER_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
+        const filterClauseFrom = new FilterClause(this.selectedField.field, this.fieldValueDateFrom, Operators.GREATER_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
         this.filters.push(filterClauseFrom);
       }
 
       if (this.fieldValueDateTo) {
-        const filterClauseTo = new FilterClause(QueryParameters.FILTER_TO, this.fieldValueDateTo, Operators.LESS_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
+        const filterClauseTo = new FilterClause(this.selectedField.field, this.fieldValueDateTo, Operators.LESS_THAN_EQUAL.value as FilterClause.Op, 'datetime', this.searchOptions);
```

Comment:
Καλή μου φαίνεται η λύση, αλλά θέλω να καταλάβω πως θα υποδεχτεί ο σερβερ. Εδώ φτιάχνεις απο 2 πεδία του UI 2 filterclauses διαφορετικά, που φαίνεται να παίρνουν το ίδιο ονομα κλειδιού-πεδίου, δηλαδή το this.selectedField.field. Τι παράγει αυτό το πράγμα και πως μπορεί να το υποδεχτεί ο σερβερ;

Reply:
Ο server δεν θα καταναλώσει άμεσα το advanced-search.component.ts. Η angular εφαρμογή θα το καταναλώσει (π.χ InPolicy.Backoffice SPA ή Cases SPA), μεταφράζοντας τα FilterClauses που δημιουργεί το library σε φίλτρα που θα περάσουν παραμετρικά στην αντίστοιχη Get server κλήση την οποία θέλουμε να φιλτράρουμε.

Η μετάφραση αυτή από την πλευρά της Angular εφαρμογής που καταναλώνει το Indice.Angular γίνεται κάπως έτσι τώρα:

```
let from = this.filters?.find(f => f.member === 'from')?.value;
let to = this.filters?.find(f => f.member === 'to')?.value;
```

όπου τα from & to περνάνε παραμετρικά στο service. 

Αυτό που προτείνω είναι κάτι τέτοιο:

```
let paymentDateFrom = this.filters?.find(f => f.member === 'paymentDate' && f.operator === Operators.GREATER_THAN_EQUAL.value as FilterClause.Op)?.value;
let paymentDateTo = this.filters?.find(f => f.member === 'paymentDate' && f.operator === Operators.LESS_THAN_EQUAL.value as FilterClause.Op)?.value;
```
ώστε να μπορείς να έχεις περισσότερα από ένα φίλτρα ημ/νιών.

Reply:
Η αλλαγή αυτή δεν επηρεάζει server APIs. Θα επηρεάσει μόνο Angular εφαρμογές που καταναλώνουν το advanced-search.component.ts της Indice.Angular βιβλιοθήκης και συγκεκριμένα τον τρόπο που διαβάζουν τα FilterClause τύπου datetime.